### ISSUE TITLE
cmd+z undo last file operation

### ISSUE THREAD

**Issue by monouser7dig** [marta-file-manager/marta-issues#230]
finder does it, I think it's very convenient and can sometimes save the day
(skimmed the keybindings but could not find it)

**Comment by yanex** [comment]
This definitely needs to be supported, though implementing "Undo" for arbitrary file operations is quite a challenging task, considering the fact that non-local file systems are also need to be supported.

**Comment by monouser7dig** [comment]
In that case I think the local 90% use case would be a good start.

**Comment by warpkanal** [comment]
Agree, local file systems and maybe undoing delete-to-trash and move would at least solve my 99% of cases.

**Comment by monouser7dig** [comment]
there is a possible edge case hiding:

If you do a local operation
then you do a remote operation

and hit cmd z; you'd expect the remote operation to be undone but that may not be implemented so the implementation should make sure to not fall back and undo the local operation that preceded the remote one

**Comment by ThSGM** [comment]
Sorry, I just wanted to write in and say that I've been using Marta the last few days, but the lack of an undo functionality might require me to drop it.
